src/diffusers/models/attention.py

```diff
@@ -190,11 +214,12 @@ def __init__(
         checkpoint: bool = True,
     ):
         super().__init__()
-        self.attn1 = CrossAttention(
+        AttentionBuilder = MemoryEfficientCrossAttention if _USE_MEMORY_EFFICIENT_ATTENTION else CrossAttention
+        self.attn1 = AttentionBuilder(
             query_dim=dim, heads=n_heads, dim_head=d_head, dropout=dropout
         )  # is a self-attention
         self.ff = FeedForward(dim, dropout=dropout, glu=gated_ff)
-        self.attn2 = CrossAttention(
```

Comment:
Could we do something like:

```python
if self._xformer_attention:
   self.attn2 = MemoryEfficientCrossAttention(
       query_dim=dim, context_dim=context_dim, heads=n_heads, dim_head=d_head, dropout=dropout
   )
else:
   self.attn2 = AttentionBuilder(
       query_dim=dim, context_dim=context_dim, heads=n_heads, dim_head=d_head, dropout=dropout
   )  # is self-attn if context is none
```

Reply:
I think introducing a new class name `AttentionBuilder` is not really necessary here - IMO it's a bit more readable to just have a if-else statement here.
`_xformer_attention` could be enabled via `enable_attention_xformers()` as described above :-)

Reply:
Added a `enable_xformers_memory_efficient_attention` method to toggle between attentions modes. Memory Efficient takes precedence over sliced_attention if it is enabled but this can be changed. 

I removed the dependency on the environment variable